### PR TITLE
[INFRA] Backport Using from Scala 2.13

### LINKED ISSUE #553: Backport Using from Scala 2.13
### 1. Describe the feature
Back port Using from Scala 2.13 to simplify `try with resources` code.
https://github.com/scala/scala/blob/v2.13.5/src/library/scala/util/Using.scala

### 2. Motivation
Scala doesn't support Java style `try with resource` in syntax. But Scala provided `Using` since 2.13 to simulate such behavior.

### 3. Describe the solution
Back port Using from Scala 2.13 into Kyuubi codebase.

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

close #553

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by turboFei — APPROVED** [review]

File: kyuubi-common/src/main/scala-2.12/scala/util/Using.scala [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+   * @param resource2 the second resource
+   * @param body      the operation to perform using the resources
+   * @tparam R1 the type of the first resource
+   * @tparam R2 the type of the second resource
+   * @tparam A  the return type of the operation
+   * @return the result of the operation, if neither the operation nor
+   *         releasing the resources throws
+   */
+  def resources[R1: Releasable, R2: Releasable, A](
+                                                    resource1: R1,
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala-2.12/scala/util/Using.scala:314]: Is scala source so ugly looking?

**Comment by yaooqinn** [comment]
can we refine one place in the main code in case of dead code?

**Comment by pan3793** [comment]
> can we refine one place in the main code in case of dead code?

Will refactor `try ... finally` by `Using.resource() { ... } ` in following PR.

File: kyuubi-common/src/main/scala-2.12/scala/util/Using.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-common/src/main/scala-2.12/scala/util/Using.scala]: updated codesytle. the original source code at https://raw.githubusercontent.com/scala/scala/v2.13.5/src/library/scala/util/Using.scala

**Review by yaooqinn — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**